general/src/datascrubbing/convert.rs

```diff
@@ -667,6 +667,30 @@ THd+9FBxiHLGXNKhG/FRSyREXEt+NyYIf/0cyByc9tNksat794ddUqnLOg0vwSkv
         assert_annotated_snapshot!(data.value().unwrap().request);
     }
 
+    #[test]
+    fn test_sanitize_http_body_string() {
+        use crate::store::StoreProcessor;
+
+        let mut data = Event::from_value(
+            serde_json::json!({
+                "request": {
+                    "data": r#"{"email":"zzzz@gmail.com","password":"zzzzz"}xxx"#
+                }
+            })
+            .into(),
+        );
+
+        // n.b.: In Rust we rely on store normalization to parse inline JSON
+
+        let mut store_processor = StoreProcessor::new(Default::default(), None);
+        process_value(&mut data, &mut store_processor, ProcessingState::root());
```

Comment:
This should no longer be necessary

Reply:
I'll keep it just to make sure we're as close as possible to what's actually happening.